sharpy/postproc/stabilityderivatives.py

```diff
@@ -0,0 +1,215 @@
+
+    def run(self, online=False):
+
+        # TODO: consider running all required solvers inside this one to keep the correct settings
+        # i.e: run Modal, Linear Assembly
+
+        derivatives = self.data.linear.derivatives  # {str:Derivatives()} (sharpy.linear.utils.derivatives.Derivatives)
+        if self.data.linear.linear_system.beam.sys.modal:
+            phi = self.data.linear.linear_system.linearisation_vectors['mode_shapes'].real
+        else:
+            phi = None
+
+        steady_forces = self.data.linear.linear_system.linearisation_vectors['forces_aero_beam_dof']
+        v0 = self.get_freestream_velocity()
+        quat = self.data.linear.tsstruct0.quat
+
+        try:
+            tpa = self.data.linear.tsstruct0.modal['t_pa']
+        except KeyError:
+            tpa = None
+
+        if self.data.linear.linear_system.uvlm.scaled:
+            raise NotImplementedError('Stability Derivatives not yet implented for scaled system')
+            self.data.linear.linear_system.update(self.settings['u_inf'])
+
+        for target_system in ['aerodynamic', 'aeroelastic']:
+            state_space = self.get_state_space(target_system)
+            target_system_derivatives = derivatives[target_system]
+
+            target_system_derivatives.initialise_derivatives(state_space,
+                                                             steady_forces,
+                                                             quat,
+                                                             v0,
+                                                             phi,
+                                                             self.data.linear.tsstruct0.modal['cg'],
+                                                             tpa=tpa)
+            target_system_derivatives.dict_of_derivatives[
+                'force_angle_velocity'] = target_system_derivatives.new_derivative(
+                'stability',
+                'angle_derivatives',
+                'Force/Angle via velocity')
+
+            # useful to double check the effect of the ``track_body`` == 'on' setting
+            # current_derivative.dict_of_derivatives['force_angle_angle'] = current_derivative.new_derivative(
+            #     'stability',
+            #     'angle_derivatives_tb',
+            #     'Force/Angle via Track Body')
+
+            target_system_derivatives.dict_of_derivatives['force_velocity'] = target_system_derivatives.new_derivative(
+                'body',
+                'body_derivatives')
+
+            target_system_derivatives.dict_of_derivatives['force_cs'] = target_system_derivatives.new_derivative(
+                'body',
+                'control_surface_derivatives')
+            target_system_derivatives.save(self.folder)
+            target_system_derivatives.savetxt(self.folder)
+
+        return self.data
+
+    def get_freestream_velocity(self):
```

Comment:
shouldn't be the same one that u_inf in settings?

Reply:
Yes, but a few quirks are needed in the case of restarted solutions, because the settings may have not been parsed and thus are still strings